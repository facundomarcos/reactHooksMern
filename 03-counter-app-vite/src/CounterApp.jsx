import PropTypes from 'prop-types';


 // props
 export const CounterApp = ({value}) =>{
   // console.log(props);
   if(!value){
    throw new Error('No existe el value');
   }

   const handleAdd = () => 
    {console.log('+1');
    value = 1000;
    console.log(value);
    }

    return(
         <>
       <h1>Counter App</h1>
        <p> { value } </p>

        <button onClick={handleAdd}> +1 </button>
   
        </>
       
    ) ;
 }
 
//se colocan al final
CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
   
}


