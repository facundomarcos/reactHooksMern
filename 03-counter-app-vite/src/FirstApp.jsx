//import { Fragment } from "react";



//fragmento, agrupa elementos 
//agrupador de elementos html en un nodo padre

// export function First(){
//     return (
//         <Fragment>
//  <h1>First App</h1>
//         <p>un parrafo</p>
//         </Fragment>
       
//     ) ;
// }


// const newMensaje = {
//     message:'Hola Mundo',
//     title: 'Facundo'
// };


// const getResult =()=>{
//     return 4+4
// }
import PropTypes from 'prop-types';


 // props
 export const First = ({title, subTitle, name}) =>{
   // console.log(props);
   if(!title){
    throw new Error('No existe el title');
   }
    return(
         <>
       <h1>{title}</h1>
        <p>{subTitle + 1}</p>
        <p>{name}</p>
        </>
       
    ) ;
 }
 
//se colocan al final
First.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

First.defaultProps = {
    name: 'Fernando',
    subTitle:'No hay subtitulo',
   // title:'No hay titulo',
    
}


