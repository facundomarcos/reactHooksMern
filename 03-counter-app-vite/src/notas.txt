https://jestjs.io/

npm install --save-dev jest


Add the following section to your package.json:

{
  "scripts": {
      "test": "jest --watchAll"
  }
}

npm run test

instalacion del intellicense
npm add --D @types/jest


configuracion babel
npm install --save-dev babel-jest @babel/core @babel/preset-env



//https://testing-library.com/
//npm install --save-dev @testing-library/react


//https://jestjs.io/docs/getting-started