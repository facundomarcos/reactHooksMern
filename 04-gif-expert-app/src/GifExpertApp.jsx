import { useState } from "react"




export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch','Dragon Ball']);

    const onAddCategory = () =>{
       // categories.push('Valorant');
      setCategories([ 'Valorant', ...categories ]);
      //setCategories(cat => [...cat, 'Valorant'])
    }

 

    console.log(categories);

  return (
    <>
    {/* titulo */}
      <h1>GifExpertApp</h1>


      {/* input */}


      {/* listado */}
      <button onClick={onAddCategory}>Agregar</button>
      <ol>
        { categories.map(categorie => {
            return <li key={ categorie }>{categorie}</li>
        })}
       
      
      </ol>
         {/* gif item */}
    </>
  )
}


