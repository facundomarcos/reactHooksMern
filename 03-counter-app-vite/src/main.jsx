import React from 'react';
import  ReactDOM  from 'react-dom/client';
import { First } from './FirstApp';
import {App} from './HelloWordApp';
import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <App/> */}
        <First/>
    </React.StrictMode>
)








