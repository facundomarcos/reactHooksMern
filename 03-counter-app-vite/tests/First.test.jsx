import {render} from '@testing-library/react';
import { First } from '../src/FirstApp';

describe('pruebas en <First />', () => {

  test('debe de hacer match con el snapshot', () => {
   const title = 'Hola, soy Goku';
    const {container}=render(<First title={title} />);

  expect(container).toMatchSnapshot();
  });
  
  test('debe mostrar el titulo un h1', () => {
    const title = 'Hola, soy Goku';
    const { container, getByText }=render(<First title={title} />);

    expect(getByText(title)).toBeTruthy();

    const h1 = container.querySelector('h1');
    expect(h1.innerHTML).toContain(title);

  })
  
});
