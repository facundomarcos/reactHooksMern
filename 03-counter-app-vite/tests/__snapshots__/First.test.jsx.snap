// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <First /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
     
    Hola, soy Goku
     
  </h1>
  <p>
    No hay subtitulo1
  </p>
  <p>
    Fernando
  </p>
</div>
`;
